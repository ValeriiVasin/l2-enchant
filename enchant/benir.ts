import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

// same chances:
// https://dev.l2central.info/essence/items/accessary/talisman/
export const benirEnchantMap = new Map([
  ...createEnchantMap({
    strategy: EnchantStrategy.Drop,
    enchantItem: Item.BenirEnchantItem,
    items: [Item.Benir_1, Item.Benir_2, Item.Benir_3, Item.Benir_4, Item.Benir_5, Item.Benir_6],
    chances: [50, 45, 40, 35, 50],
  }),
  ...createEnchantMap({
    strategy: EnchantStrategy.Drop,
    enchantItem: Item.BenirEnchantItem,
    items: [Item.Benir_6, Item.Benir_7, Item.Benir_8, Item.Benir_9, Item.Benir_10, Item.Benir_11, Item.Benir_12],
    chances: [45, 40, 35, 30, 25, 50],
  }),
  ...createEnchantMap({
    strategy: EnchantStrategy.Drop,
    enchantItem: Item.BenirEnchantItem,
    items: [Item.Benir_12, Item.Benir_13, Item.Benir_14, Item.Benir_15, Item.Benir_16, Item.Benir_17, Item.Benir_18],
    chances: [35, 30, 25, 20, 15, 50],
  }),
  ...createEnchantMap({
    strategy: EnchantStrategy.Drop,
    enchantItem: Item.BenirEnchantItem,
    items: [Item.Benir_18, Item.Benir_19, Item.Benir_20, Item.Benir_21, Item.Benir_22, Item.Benir_23, Item.Benir_24],
    chances: [35, 30, 25, 20, 15, 50],
  }),
  ...createEnchantMap({
    strategy: EnchantStrategy.Drop,
    enchantItem: Item.BenirEnchantItem,
    items: [Item.Benir_24, Item.Benir_25],
    chances: [35],
  }),
]);
