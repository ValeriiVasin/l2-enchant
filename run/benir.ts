import { run } from '../helpers/run';
import { Item } from '../items';

run(
  Item.Benir_1,
  [
    Item.Benir_2,
    Item.Benir_3,
    Item.Benir_4,
    Item.Benir_5,
    Item.Benir_6,
    Item.Benir_7,
    Item.Benir_8,
    Item.Benir_9,
    Item.Benir_10,
    Item.Benir_11,
    Item.Benir_12,
    Item.Benir_13,
    Item.Benir_14,
    Item.Benir_15,
    Item.Benir_16,
    Item.Benir_17,
    Item.Benir_18,
    Item.Benir_19,
    Item.Benir_20,
    Item.Benir_21,
    Item.Benir_22,
    Item.Benir_23,
    Item.Benir_24,
    Item.Benir_25,
  ],
  100_000,
);
