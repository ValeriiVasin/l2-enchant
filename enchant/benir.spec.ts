import { benirEnchantMap } from './benir';

it('enchant map: benir', () => {
  expect(benirEnchantMap).toMatchSnapshot();
});
