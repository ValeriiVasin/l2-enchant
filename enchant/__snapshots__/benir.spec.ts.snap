// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: benir 1`] = `
Map {
  "Талисман Бенира Ур.1" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.1",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.2",
      },
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.1",
      },
    ],
  },
  "Талисман Бенира Ур.2" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.2",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Талисман Бенира Ур.3",
      },
      Object {
        "chance": 55,
        "item": "Талисман Бенира Ур.1",
      },
    ],
  },
  "Талисман Бенира Ур.3" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.3",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Бенира Ур.4",
      },
      Object {
        "chance": 60,
        "item": "Талисман Бенира Ур.1",
      },
    ],
  },
  "Талисман Бенира Ур.4" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.4",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Бенира Ур.5",
      },
      Object {
        "chance": 65,
        "item": "Талисман Бенира Ур.1",
      },
    ],
  },
  "Талисман Бенира Ур.5" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.5",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.6",
      },
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.1",
      },
    ],
  },
  "Талисман Бенира Ур.6" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.6",
    "results": Array [
      Object {
        "chance": 45,
        "item": "Талисман Бенира Ур.7",
      },
      Object {
        "chance": 55,
        "item": "Талисман Бенира Ур.6",
      },
    ],
  },
  "Талисман Бенира Ур.7" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.7",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Талисман Бенира Ур.8",
      },
      Object {
        "chance": 60,
        "item": "Талисман Бенира Ур.6",
      },
    ],
  },
  "Талисман Бенира Ур.8" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.8",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Бенира Ур.9",
      },
      Object {
        "chance": 65,
        "item": "Талисман Бенира Ур.6",
      },
    ],
  },
  "Талисман Бенира Ур.9" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.9",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Бенира Ур.10",
      },
      Object {
        "chance": 70,
        "item": "Талисман Бенира Ур.6",
      },
    ],
  },
  "Талисман Бенира Ур.10" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.10",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Талисман Бенира Ур.11",
      },
      Object {
        "chance": 75,
        "item": "Талисман Бенира Ур.6",
      },
    ],
  },
  "Талисман Бенира Ур.11" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.11",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.12",
      },
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.6",
      },
    ],
  },
  "Талисман Бенира Ур.12" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.12",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Бенира Ур.13",
      },
      Object {
        "chance": 65,
        "item": "Талисман Бенира Ур.12",
      },
    ],
  },
  "Талисман Бенира Ур.13" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.13",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Бенира Ур.14",
      },
      Object {
        "chance": 70,
        "item": "Талисман Бенира Ур.12",
      },
    ],
  },
  "Талисман Бенира Ур.14" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.14",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Талисман Бенира Ур.15",
      },
      Object {
        "chance": 75,
        "item": "Талисман Бенира Ур.12",
      },
    ],
  },
  "Талисман Бенира Ур.15" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.15",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Талисман Бенира Ур.16",
      },
      Object {
        "chance": 80,
        "item": "Талисман Бенира Ур.12",
      },
    ],
  },
  "Талисман Бенира Ур.16" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.16",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Талисман Бенира Ур.17",
      },
      Object {
        "chance": 85,
        "item": "Талисман Бенира Ур.12",
      },
    ],
  },
  "Талисман Бенира Ур.17" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.17",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.18",
      },
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.12",
      },
    ],
  },
  "Талисман Бенира Ур.18" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.18",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Бенира Ур.19",
      },
      Object {
        "chance": 65,
        "item": "Талисман Бенира Ур.18",
      },
    ],
  },
  "Талисман Бенира Ур.19" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.19",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Талисман Бенира Ур.20",
      },
      Object {
        "chance": 70,
        "item": "Талисман Бенира Ур.18",
      },
    ],
  },
  "Талисман Бенира Ур.20" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.20",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Талисман Бенира Ур.21",
      },
      Object {
        "chance": 75,
        "item": "Талисман Бенира Ур.18",
      },
    ],
  },
  "Талисман Бенира Ур.21" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.21",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Талисман Бенира Ур.22",
      },
      Object {
        "chance": 80,
        "item": "Талисман Бенира Ур.18",
      },
    ],
  },
  "Талисман Бенира Ур.22" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.22",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Талисман Бенира Ур.23",
      },
      Object {
        "chance": 85,
        "item": "Талисман Бенира Ур.18",
      },
    ],
  },
  "Талисман Бенира Ур.23" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.23",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.24",
      },
      Object {
        "chance": 50,
        "item": "Талисман Бенира Ур.18",
      },
    ],
  },
  "Талисман Бенира Ур.24" => Object {
    "cost": "Осколок Талисмана Бенира",
    "item": "Талисман Бенира Ур.24",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Талисман Бенира Ур.25",
      },
      Object {
        "chance": 65,
        "item": "Талисман Бенира Ур.24",
      },
    ],
  },
}
`;
